cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/PartitionedFileSet.java

```diff
@@ -93,6 +93,14 @@
   void dropPartition(PartitionKey key);
 
   /**
+   * Concatenates the partition in Hive. If this Dataset is not enabled for Explore, this operation is a no-op.
```

Comment:
I made the PartitionedFileSet#concatenatePartition(PartitionKey) method async since it can take a while by running a MapReduce job.